index.js

```diff
@@ -153,8 +168,16 @@ NYC.prototype._wrapRequire = function () {
 
     _this.sourceMapCache.add(filename, code)
 
-    // now instrument the compiled code.
-    return _this.instrumenter.instrumentSync(code, './' + relFile)
+    var hash = md5(code)
+    var cacheFilePath = path.join(_this._cacheDirectory(), hash + '.js')
+
+    try {
+      return fs.readFileSync(cacheFilePath, 'utf8')
+    } catch (e) {
+      var instrumented = _this.instrumenter().instrumentSync(code, './' + relFile)
+      fs.writeFile(cacheFilePath, instrumented)
```

Comment:
> I suppose this could use async with a `console.warn()` if there are errors?

I think we are going to run into a lot more write collisions when they start using `nyc` with `AVA`. My tests to this point have been the `AVA` test suite (which is run with TAP). I think we only need to warn on `read` errors.

Perhaps we could log `write` errors with [`debug`](https://github.com/visionmedia/debug).